File: OnlineBookReader/src/Person.cpp
#include "Person.h"


Person :: Person(){
}
Person :: Person(const string &username ,const string &password ,const string &name ,const string &email){
    this->username = username ;
    this->password = password ;
    this->name = name ;
    this->email = email ;
}
Person :: Person(string& line){
    vector<string> words = Split_string(line) ;
    assert(words.size() == 4) ;
    this->username = words[0] ;
    this->password = words[1] ;
    this->name = words[2] ;
    this->email = words[3] ;
}
void Person :: setUsername(const string& username){
    this->username = username ;
}
const string& Person :: getUsername() const{
    return username ;
}
void Person :: setName(const string& name){
    this->name = name ;
}
const string& Person :: getName() const{
    return name ;
}
void Person :: setPassword(const string& password){
    this->password = password ;
}
const string& Person :: getPassword() const{
    return password ;
}
void Person:: setEmail(const string& email){
    this->email = email ;
}
const string& Person :: getEmail() const{
    return email ;
}
void Person :: to_string() const{
    cout << "Name: " << name ;
    cout << "\nEmail: " << email ;
    cout << "\nUser name: " << username ;
    cout << "\npassword : " << password ;
}

File: ManagementOfStudentAbsences/src/managementOfStudentAbsences/Person.java
package managementOfStudentAbsences;

public class Person {
	protected String name;
	protected String surname; 
	protected String username ; 
	protected String password ; 
	Person(){
		
	}
	Person(String name , String surname , String username, String password){
		this.name = name ; 
		this.surname = surname ; 
		this.username = username ; 
		this.password = password ; 
	}
	protected void setPassword(String password) {
		this.password = password ; 
	}
	protected String getPassword() {
		return this.password ; 
	}
	protected void setName(String name) {
		this.name = name ; 
	}
	protected String getName() {
		return name ; 
	}
	protected void setSurname(String surname) {
		this.surname = surname ; 
	}
	protected String getSurname() {
		return surname ; 
	}
	protected void setUsername(String username) {
		this.username = username ; 
	}
	protected String getUsername() {
		return username ; 
	}
}

